### PR TITLE
Comet update

### PR DESCRIPTION
**Pull request by Ryang-21**
- Updated comet wasm and comet test setup
- Updated comet authorization in backstop claim function
- Waiting on PR CometDEX/comet-contracts-v1#5

### CODE REVIEW

File: pool/src/storage.rs
Comment (mootz12) [inline pool/src/storage.rs:350]: was this a typo / test additional that did not get reverted?

File: pool/src/storage.rs
Comment (mootz12) [inline pool/src/storage.rs:367]: :ditto:

File: pool/src/storage.rs
Comment (mootz12) [inline pool/src/storage.rs:441]: :ditto:

File: pool/src/storage.rs
Comment (mootz12) [inline pool/src/storage.rs:458]: :ditto:

File: pool/src/testutils.rs
Comment (mootz12) [inline pool/src/testutils.rs:174]: why was this change necessary?

IIRC `finalize` had nothing to do with `set_public_swap`, but this could have changed on comet's end?

**Review by mootz12 — CHANGES_REQUESTED** [review]
Need to fix the rent bump changes for reserve config and data.

I would also like to put this PR on hold until your Comet changes get merged. Would you mind linking that PR here @Ryang-21 ?

File: pool/src/storage.rs
Comment (Ryang-21) [reply pool/src/storage.rs]: Yeah forgot to revert from my ledger restore pr. This has been fixed

**Review by mootz12 — APPROVED** [review]
LG2M

**Comment by mootz12** [comment]
Merging for now as the Comet PR is still being reviewed. It appears likely this will go through, so in the event something changes, we will do a followup PR.

**Pull request merged**
